core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -919,6 +919,7 @@ public void sendRequest(final DiscoveryNode node, final long requestId, final St
                     transportServiceAdapter.onRequestSent(node, requestId, action, request, finalOptions);
                 }
             };
+            Channel targetChannel = nodeChannel(node, options);
```

Comment:
I wanted to reduce the scope (or rather the lifetime) of this variable and bring it as close as possible to its usage. It's not needed anywhere before this point and t.b.h. we can even inline it and get rid of the variable.